### PR TITLE
Ensure target imagestream exists before promoting

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @smarterclayton

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@smarterclayton ping

File: pkg/steps/promote.go [hunk 1]
```diff
@@ -102,9 +102,25 @@ func (s *promotionStep) Run(ctx context.Context, dry bool) error {
 		if valid == nil {
 			continue
 		}
+
+		name := fmt.Sprintf("%s%s", s.config.NamePrefix, dst)
+
+		_, err := s.dstClient.ImageStreams(s.config.Namespace).Get(name, meta.GetOptions{})
```
Comment (smarterclayton) [inline pkg/steps/promote.go]: You can just do a blind create and ignore already exists.  That would be simpler and easier to reason about.

Comment (smarterclayton) [reply pkg/steps/promote.go]: And ignore forbidden (because you might have update permission but not create).

Comment (stevekuznetsov) [reply pkg/steps/promote.go]: ACK

**Comment by stevekuznetsov** [comment]
@smarterclayton actually no I thought about it and I disagree. If we see there is no imagestream, we should create one. If we cannot create one and we know one does not exist, we should fail. No job asking for `--promote` should succeed if the output image is not present, as is currently the case ...

**Comment by smarterclayton** [comment]
Promotion is either going to succeed or fail.  The only permission ci-operator requires is the ability to update a tag.  If you don’t have create permissions and the stream doesn’t exist, the tag update will fail with the correct error.  If you do have permission to update and the stream exists, then you didn’t need create permission.

There’s no need to check the get before calling create.

**Comment by smarterclayton** [comment]
Yeah, still don't agree with this change.

**Pull request merged**